api/src/main/java/org/openmrs/module/kenyaemr/advice/SyncHFEAppointmentsWithBahmniModule.java

```diff
@@ -98,183 +117,275 @@ public void afterReturning(Object returnValue, Method method, Object[] args, Obj
+
+            if(editAppointment != null) {
+                //edit MCH or PREP or TB or KP appointment
+
+                Date nextApptStartDateTime = DateUtil.convertToDate(dateFormat.format(o.getValueDatetime()).concat("T07:00:00.0Z"), DateUtil.DateFormatType.UTC);
+                Date nextApptEndDateTime = DateUtil.convertToDate(dateFormat.format(o.getValueDatetime()).concat("T20:00:00.0Z"), DateUtil.DateFormatType.UTC);
+                editAppointment.setStartDateTime(nextApptStartDateTime);
+                editAppointment.setEndDateTime(nextApptEndDateTime);
+                Appointment app = appointmentsService.validateAndSave(editAppointment);
+
+            } else {
+                // create MCH or TB or KP or PREP appointment
+                if (enc.getForm() != null && enc.getForm().getUuid().equals(MchMetadata._Form.MCHMS_POSTNATAL_VISIT) || enc.getForm() != null && enc.getForm().getUuid().equals(MchMetadata._Form.MCHMS_DELIVERY)) {
+                    appointmentServiceDefinition.setAppointmentServiceId(appointmentServiceDefinitionService.getAppointmentServiceByUuid(MCH_POSTNATAL_VISIT_SERVICE).getId());
+                } else if (enc.getForm() != null && enc.getForm().getUuid().equals(MchMetadata._Form.MCHMS_ANTENATAL_VISIT)) {
+                    appointmentServiceDefinition.setAppointmentServiceId(appointmentServiceDefinitionService.getAppointmentServiceByUuid(MCH_ANTENATAL_VISIT_SERVICE).getId());
+                } else if (enc.getForm() != null && enc.getForm().getUuid().equals(MchMetadata._Form.MCHCS_FOLLOW_UP)) {
+                    appointmentServiceDefinition.setAppointmentServiceId(appointmentServiceDefinitionService.getAppointmentServiceByUuid(CWC_FOLLOWUP_SERVICE).getId());
+                } else if (enc.getForm() != null && enc.getForm().getUuid().equals(PREP_INITIAL_FORM)) {
+                    appointmentServiceDefinition.setAppointmentServiceId(appointmentServiceDefinitionService.getAppointmentServiceByUuid(PREP_INITIAL_SERVICE).getId());
+                } else if (enc.getForm() != null && enc.getForm().getUuid().equals(PREP_FOLLOWUP_FORM)) {
+                    appointmentServiceDefinition.setAppointmentServiceId(appointmentServiceDefinitionService.getAppointmentServiceByUuid(PREP_FOLLOWUP_SERVICE).getId());
+                } else if (enc.getForm() != null && enc.getForm().getUuid().equals(PREP_MONTHLY_REFILL_FORM)) {
+                    appointmentServiceDefinition.setAppointmentServiceId(appointmentServiceDefinitionService.getAppointmentServiceByUuid(PREP_MONTHLY_REFILL_SERVICE).getId());
+                } else if (enc.getForm() != null && enc.getForm().getUuid().equals(TbMetadata._Form.TB_FOLLOW_UP)) {
+                    appointmentServiceDefinition.setAppointmentServiceId(appointmentServiceDefinitionService.getAppointmentServiceByUuid(TB_SERVICE).getId());
+                } else if (enc.getForm() != null && enc.getForm().getUuid().equals(KP_CLINICAL_VISIT_FORM)) {
```

Comment:
I think on the if statement you can add a check for each service to ensure it is not null or else if it tries to get id and it is not there it will throw.

Reply:
And also when creating appointments ensure the service Id we are setting is not null.